lib/containers/container_images.pm

```diff
@@ -123,23 +138,55 @@ sub test_opensuse_based_image {
             my $pretty_version = $version =~ s/-SP/ SP/r;
             my $betaversion    = get_var('BETA') ? '\s\([^)]+\)' : '';
             record_info "Validating", "Validating That $image has $pretty_version on /etc/os-release";
-            validate_script_output("$runtime container run --entrypoint '/bin/bash' --rm $image -c 'grep PRETTY_NAME /etc/os-release' | cut -d= -f2",
-                sub { /"SUSE Linux Enterprise Server ${pretty_version}${betaversion}"/ });
+            if ($runtime =~ /docker/) {
+                # zypper-docker changes the layout of the image
+                validate_script_output("$runtime run --entrypoint /bin/bash $image -c 'grep PRETTY_NAME /etc/os-release' | cut -d= -f2",
+                    sub { /"SUSE Linux Enterprise Server ${pretty_version}${betaversion}"/ });
+            }
+            else {
+                validate_script_output("$runtime run $image grep PRETTY_NAME /etc/os-release | cut -d= -f2",
+                    sub { /"SUSE Linux Enterprise Server ${pretty_version}${betaversion}"/ });
+            }
+
             # SUSEConnect zypper service is supported only on SLE based image on SLE host
             my $plugin = '/usr/lib/zypp/plugins/services/container-suseconnect-zypp';
-            assert_script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin -v'";
-            script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin lp'", 420;
-            script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin lm'", 420;
-        } else {
+            if ($runtime =~ /buildah/) {
+                assert_script_run "$runtime run -t $image -- $plugin -v";
+                script_run "$runtime run -t $image -- $plugin lp", 420;
+                script_run "$runtime run -t $image -- $plugin lm", 420;
+            } else {
+                assert_script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin -v'";
+                script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin lp'", 420;
+                script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin lm'", 420;
+            }
+        }
+        else {
             record_info "non-SLE host", "This host ($host_id) does not support zypper service";
         }
-    } else {
+    }
+    else {
         $version =~ s/^Jump://i;
         validate_script_output qq{$runtime container run --entrypoint '/bin/bash' --rm $image -c 'cat /etc/os-release'}, sub { /PRETTY_NAME="openSUSE (Leap )?${version}.*"/ };
     }
 
     # Zypper is supported only on openSUSE or on SLE based image on SLE host
     if (($host_id =~ 'sles' && $image_id =~ 'sles') || $image_id =~ 'opensuse') {
+        test_zypper_on_container($runtime, $image);
+    }
+}
+
+sub test_zypper_on_container {
+    my ($runtime, $image) = @_;
+    if ($runtime =~ /buildah/) {
```

Comment:
There is in my opinion no need for this condition…

Reply:
there are a few differences. how do you want to workaround them?

Reply:
OK, I see now :) But the first command can be excluded from the condition as the `--rm` is not needed:
```perl
# zypper lr
assert_script_run("$runtime run $image zypper lr -s", 120);
```

Reply:
done 946e17650